packages/core/utils/src/modules-sdk/export-module.ts

```diff
@@ -0,0 +1,44 @@
+import { Constructor, ModuleExports } from "@medusajs/types"
+import { MedusaServiceDmlObjectSymbolFunction } from "./medusa-service"
+import {
+  buildLinkConfigFromDmlObjects,
+  defineJoinerConfig,
+} from "./joiner-config-builder"
+import { InfersLinksConfig } from "./types/links-config"
+import { DmlEntity } from "../dml"
+
+/**
+ * Build the module export and auto generate the joiner config if needed as well as
+ * return a links object based on the DML objects
+ * @param moduleName
+ * @param service
+ * @param loaders
+ * @constructor
+ */
+export function ExportModule<
```

Comment:
suggestion: Should we call this `MedusaModule` so it is similar to `MedusaService`? Imo. we don't need the `Export` keyword to be part of the function definition. That is the action you are taking when using it. 
```suggestion
export function MedusaModule<
```

Reply:
`MedusaModule` will conflict with something that we already have. If this is choosen, we have to update the current one.
But I like the name. or simply `Module`

```typescript
export default Module(Modules.CURRENCY, {
  service,
  loaders,
})
```

Reply:
Ah good point, I think `Module` works just as well

Reply:
I ll rename it then 💪